src/EnergyPlus/ChillerElectricASHRAE205.cc

```diff
@@ -0,0 +1,1665 @@
+    const Real64 maximumChillerCap = this->Representation->performance.performance_map_cooling
+                                         .calculate_performance(this->EvapVolFlowRate,
+                                                                this->EvapOutletTemp + DataGlobalConstants::KelvinConv,
+                                                                this->CondVolFlowRate,
+                                                                this->CondInletTemp + DataGlobalConstants::KelvinConv,
+                                                                this->MaxSequenceNumber)
+                                         .net_evaporator_capacity;
+    if (maximumChillerCap <= 0) {
+        // TODO: Issue a warning
+    }
+    const Real64 minimumChillerCap = this->Representation->performance.performance_map_cooling
+                                         .calculate_performance(this->EvapVolFlowRate,
+                                                                this->EvapOutletTemp + DataGlobalConstants::KelvinConv,
+                                                                this->CondVolFlowRate,
+                                                                this->CondInletTemp + DataGlobalConstants::KelvinConv,
+                                                                this->MinSequenceNumber)
+                                         .net_evaporator_capacity;
+    // Part load ratio based on load and available chiller capacity; cap at max P.L.R. (can be >1)
+    this->ChillerPartLoadRatio = (maximumChillerCap > 0) ? max(0.0, std::abs(MyLoad) / maximumChillerCap) : 0.0;
+    // Minimum capacity ratio, under which cycling occurs
+    this->MinPartLoadRat = (maximumChillerCap > 0) ? minimumChillerCap / maximumChillerCap : 0.0;
+    Real64 partLoadSeqNum{0.};
+
+    // Chiller may be operating in one of three modes: cycling, modulating, or full capacity
+    if (this->ChillerPartLoadRatio < this->MinPartLoadRat) // Cycling
+    {
+        this->ChillerCyclingRatio = this->ChillerPartLoadRatio / this->MinPartLoadRat;
+        partLoadSeqNum = this->MinSequenceNumber;
+    } else if (this->ChillerPartLoadRatio < 1.0) // Modulating
+    {
+        // Use performance map to find the fractional sequence number (which most closely matches our part load)
+        auto f = std::bind(&ASHRAE205ChillerSpecs::findCapacityResidual, this, std::placeholders::_1, std::placeholders::_2, std::placeholders::_3);
+        Real64 constexpr accuracy{0.0001};
+        int constexpr maxIter{500};
+        int solFla{0};
+        std::array<Real64, 4> par{{MyLoad, RunFlag ? 1.0 : 0.0, 1.0}}; // Initialize iteration parameters for RegulaFalsi function
+        // Iteratively calculate this->QEvaporator by modulating partLoadSeqNum, ending at Q_Evaporator(partLoadSeqNum)
+        General::SolveRoot(state, accuracy, maxIter, solFla, partLoadSeqNum, f, this->MinSequenceNumber, this->MaxSequenceNumber, par);
```

Comment:
SolveRoot actually takes a `Function` and generic payload [now](https://github.com/NREL/EnergyPlus/blob/a1211bbb5613f31be3bc6b789d48652b956b8005/src/EnergyPlus/General.hh#L85), but we've not really exercised it that way.  Would have been a nice opportunity to use a lambda and generic payload on some new modern C++ code in here, but oh well.

Reply:
Bummer. Next time, or a subsequent revision then?
I also wonder if the template could use SFINAE rather than static_assert? (idle thought)

Reply:
Next time.